analytics/analytics/core/job_failure_classifier/__init__.py

```diff
@@ -132,13 +177,18 @@ def upload_job_failure_classification(job_input_data_json: str) -> None:
 
     # Read input data and extract params
     job_input_data = json.loads(job_input_data_json)
-    job_id = job_input_data["build_id"]
-    job_name = job_input_data["build_name"]
 
     # Annotate if job has been retried
-    attempt_number, retried = _job_retry_data(job_id=job_id, job_name=job_name)
-    job_input_data["attempt_number"] = attempt_number
-    job_input_data["retried"] = retried
+    try:
+        retry_info = _job_retry_data(
+            job_id=job_input_data["build_id"],
+            job_name=job_input_data["build_name"],
+            job_commit_id=job_input_data["commit"]["id"],
+            job_failure_reason=job_input_data["build_failure_reason"],
+        )
+        job_input_data.update(asdict(retry_info))
+    except Exception:
+        sentry_sdk.capture_exception()
```

Comment:
I tested this locally but this is just out of an abundance of caution for the initial deployment. Assuming it's not triggered within the first few days we can remove it.